track/log_manager.py

```
@@ -0,0 +1,73 @@
+import logging
+
+from django_apscheduler.models import DjangoJobExecution
+
+from track.models import *
+
+logger = logging.getLogger(__name__)
+
+
+class LogManager:
+
+    def start_order_log(self, order_number):
+        existing_log = OrderLog.objects.filter(order_number=order_number, is_complete=False).first()
+        if existing_log:
+            self.complete_log(order_number)
+        OrderLog.objects.create(order_number=order_number)
+
+    def start_confirmation_log(self):
+        existing_log = ConfirmationCheckLog.objects.filter(is_complete=False).first()
+        if existing_log:
+            self.complete_log()
+        job_id = DjangoJobExecution.objects.latest('run_time').id
+        ConfirmationCheckLog.objects.create(job_id=job_id)
+
+    def get_log(self, order_number=None):
+        log = None
+        if order_number:
+            try:
+                return OrderLog.objects.get(order_number=order_number, is_complete=False)
+            except Exception as e:
+                logger.error(f"{inspect.stack()[0][3]}: {e}")
+                logger.error(f"{inspect.stack()[0][3]}: OrderLog not found.")
```

there needs to be an explicit return statement either here or at the end of the method.

---

can i just raise a `DoesNotExist` exception here instead?

---

i don't really know what I'd return. My thinking was that because so many methods use this method to grab the log, so I should handle all of the exceptions here

---

you can either raise an exception or return None. either is probably fine as long as it doesn't interrupt the order flow (the order should still go through even if the log can't be retrieved).